### PR TITLE
[Feat] Search and Filter Base Review Page

### PR DESCRIPTION
**Pull request by jessiiepeng**
## Notion ticket link

[Search and Filter Base Review Page](https://www.notion.so/uwblueprintexecs/16fb020b93f641c8b384f76a1b3c769e?v=aedc8d99cd8b437cb556ece10f3c0e3c&p=169bc7491da340a9876892d52841247f)

## Implementation description
* Store state of url params and search query in search bar
* Put review grid and search bar together
* OMIT: filters component because not ready yet -> new ticket here: https://www.notion.so/uwblueprintexecs/16fb020b93f641c8b384f76a1b3c769e?v=aedc8d99cd8b437cb556ece10f3c0e3c&p=db301040f994483388abec0906627633&pm=c

## Steps to test
1. Go to `http://localhost:3000/magazine/search_results`
2. search something in the search bar and verify that it updates the url
3. http://localhost:3000/magazine/search_results?search_query=RANDOM TEXT HERE` and verify that the searchbar is updated
4. verify that page is responsive by resizing

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have run docker-compose up and my project compiled
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/SearchReviews/SearchReviews.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+const SearchReviews = (): React.ReactElement => {
+  const [loading, setLoading] = useState<boolean>(false);
+  const [searchText, setSearchText] = useState<string>("");
+  const [genresFilter, setGenresFilter] = useState<string[]>([]);
+  const [ageRangeFilter, setAgeRangeFilter] = useState<number[]>([]); // ageRange[0] is min age, ageRange[1] is max age
+  const [displayedReviews, setDisplayedReviews] = useState<Review[]>([]);
+
+  /** Fetches the search queries from url */
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search);
```
Comment (susheelk) [inline frontend/src/components/pages/SearchReviews/SearchReviews.tsx:22]: Could you document these search params? (ie what each of them are, what they do, some examples, etc). You could maybe put it in the component header

File: frontend/src/components/pages/SearchReviews/SearchReviews.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+  }, []);
+
+  /** Changes url to when search filters are changed and fetches search results  */
+  useEffect(() => {
+    setLoading(true);
+    // todo add age range filter too
+    let newSearchUrl = "/magazine/search_results";
+    newSearchUrl +=
+      searchText || genresFilter.length > 0 || ageRangeFilter.length > 0
```
Comment (susheelk) [inline frontend/src/components/pages/SearchReviews/SearchReviews.tsx:50]: You might run into weird issues with embedding (ie when using special characters, etc) when building the URL like this. I'd suggest looking into JS's built in url library: https://www.valentinog.com/blog/url/

File: frontend/src/components/pages/SearchReviews/SearchReviews.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    if (intAgeRange) {
+      setAgeRangeFilter(intAgeRange);
+    }
+  }, []);
+
+  /** Changes url to when search filters are changed and fetches search results  */
+  useEffect(() => {
+    setLoading(true);
+    // todo add age range filter too
+    let newSearchUrl = "/magazine/search_results";
```
Comment (susheelk) [inline frontend/src/components/pages/SearchReviews/SearchReviews.tsx:48]: I would suggest moving this URL generation code into a separate helper function (ie one that takes search text, filters as arguments and returns a generated URL). In the future, we'll have other parts of our code also generating URLs and redirecting to this page, and such a function would be useful then.

**Review by susheelk — CHANGES_REQUESTED** [review]
Nice work! The url changing seems to work as desired :slightly_smiling_face: 

Just a couple of minor comments:
* This wasn't part of this ticket, but hitting enter on the search bar doesn't submit. Could you please add this to your PR?
* There doesn't seem to be any spacing at the bottom of the page:
image
We probably want to add some for readability 

Thanks!

File: frontend/src/components/pages/SearchReviews/SearchReviews.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+import { Review } from "../../../types/ReviewTypes";
+import LoadingSpinner from "../../common/LoadingSpinner";
+import SearchBox from "../SearchBox";
+import mockReviews from "./mockReviews";
+import ReviewsGrid from "./ReviewsGrid";
+
+/**
+ * The component for the page where the user searches and filters reviews.
+ * Url Parameters that can be passed in:
```
Comment (susheelk) [inline frontend/src/components/pages/SearchReviews/SearchReviews.tsx:12]: (nit) Indent this

File: frontend/src/components/pages/SearchReviews/SearchReviews.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+ * search_query: current search command i.e "Potter"
```
Comment (susheelk) [inline frontend/src/components/pages/SearchReviews/SearchReviews.tsx:13]: (nit) Double indent these

**Review by susheelk — APPROVED** [review]
LGTM! :slightly_smiling_face: 

Go ahead and merge after resolving these nitpicks. Also could you make another ticket to add the background image to this page? Thanks!

**Comment by jessiiepeng** [comment]
> LGTM! 🙂
> 
> Go ahead and merge after resolving these nitpicks. Also could you make another ticket to add the background image to this page? Thanks!

https://www.notion.so/uwblueprintexecs/16fb020b93f641c8b384f76a1b3c769e?v=aedc8d99cd8b437cb556ece10f3c0e3c&p=130533dc9d8a462b824114f5d761b348&pm=c
Ticket ^
Should I assign these the adding filtering component / background images tickets to myself?

**Pull request merged**